GetEycaActivation/handler.ts

```diff
@@ -0,0 +1,142 @@
+    orchestratorStatus.customStatus === "COMPLETED"
+  ) {
+    return taskEither.of(StatusEnum.COMPLETED);
+  }
+  switch (orchestratorStatus.runtimeStatus) {
+    case df.OrchestrationRuntimeStatus.Pending:
+      return taskEither.of(StatusEnum.PENDING);
+    case df.OrchestrationRuntimeStatus.Running:
+    case df.OrchestrationRuntimeStatus.ContinuedAsNew:
+      return taskEither.of(StatusEnum.RUNNING);
+    case df.OrchestrationRuntimeStatus.Failed:
+      return taskEither.of(StatusEnum.ERROR);
+    case df.OrchestrationRuntimeStatus.Completed:
+      return taskEither.of(StatusEnum.COMPLETED);
+    default:
+      return fromLeft(
+        ResponseErrorNotFound("Not found", "Cannot recognize status")
+      );
+  }
+};
+
+export function GetEycaActivationHandler(
+  userEycaCardModel: UserEycaCardModel
+): IGetEycaActivationHandler {
+  return async (context, fiscalCode) => {
+    const client = df.getClient(context);
+    const orchestratorId = makeEycaOrchestratorId(
+      fiscalCode,
+      StatusEnum.PENDING
+    );
+    // first check if an activation process is running
+    return retrieveUserEycaCard(userEycaCardModel, fiscalCode)
+      .map(_ => _.card)
+      .chain(eycaCard =>
+        getOrchestratorStatus(client, orchestratorId)
+          .mapLeft<IResponseErrorInternal | IResponseErrorNotFound>(() =>
+            ResponseErrorInternal("Cannot retrieve activation status")
+          )
+          .chain<EycaActivationDetail>(maybeOrchestrationStatus =>
+            fromNullable(maybeOrchestrationStatus).foldL(
+              () =>
+                fromLeft(
+                  ResponseErrorNotFound(
+                    "Cannot find any activation process",
+                    "Orchestrator instance not found"
+                  )
+                ),
+              orchestrationStatus =>
+                // now try to map orchestrator status
+                mapOrchestratorStatus(orchestrationStatus).map(_ => ({
+                  created_at: orchestrationStatus.createdTime,
+                  last_updated_at: orchestrationStatus.lastUpdatedTime,
+                  status: _
+                }))
+            )
+          )
+          .foldTaskEither<
+            IResponseErrorInternal | IResponseErrorNotFound,
+            EycaActivationDetail
+          >(
+            () =>
+              // It's not possible to map any activation status
+              // check for EYCA CArd status on cosmos
+              taskEither
+                .of<
+                  IResponseErrorInternal | IResponseErrorNotFound,
+                  StatusEnum
+                >(
+                  CardActivated.is(eycaCard)
+                    ? StatusEnum.COMPLETED
+                    : StatusEnum.PENDING
+                )
+                .map(_ => ({ status: _ })),
+            activationDetail => taskEither.of(activationDetail)
+          )
+      )
+      .fold<ResponseTypes>(identity, _ => ResponseSuccessJson(_))
```

Comment:
what's the point of using a specific response error and then respond with a default status? Just use a generic value (`Error` is fine, but it may be whatever).
This whole chunk can be
```ts
.fold<ResponseTypes>(
  _ => ({
    status: CardActivated.is(eycaCard) ? StatusEnum.COMPLETED : StatusEnum.PENDING
  }),
  _ => ResponseSuccessJson(_)
)
```

Also, shouldn't left branch return a response type as well?

Reply:
You're right i'd rather move `foldTaskEither`

Reply:
Since this `foldTaskEither` is used essentially to map left cases of `checkOrchestratorStatus` I'd rather keep it in this way. The motivation of such usage is:
- An activation status could be commonly inferred by the status of the model on cosmos (and if it fails we should return proper Response( Not Found, Internal )
- if orchestrator status check fails it cannot be mapped as an Error since the orchestrator status is unknown and the status on cosmos model is `PENDING`, we consider it as RUNNING (or PENDING indeed).
I will map left as `Error` in that cases, actually it can generate confusion :)